base/regex.jl

```diff
@@ -152,8 +152,10 @@ function match(re::Regex, str::UTF8String, idx::Integer, add_opts::UInt32=UInt32
     RegexMatch(mat, cap, ovec[1]+1, off, re)
 end
 
-match(re::Regex, str::Union{ByteString,SubString}, idx::Integer, add_opts::UInt32=UInt32(0)) =
-    match(re, utf8(str), idx, add_opts)
+_utf8(str) = utf8(str)
```

Comment:
@JeffBezanson, see the discussion [here](https://github.com/JuliaLang/julia/pull/12265#discussion_r35243081).